mne/tests/test_annotations.py

```diff
@@ -468,4 +468,199 @@ def test_annotations_crop():
         a.copy().crop(tmin=0, tmax=12, emit_warning=True)
 
 
+@testing.requires_testing_data
+def test_events_from_annot_in_raw_objects():
+    """Test basic functionality of events_fron_annot for raw objects."""
+    raw = read_raw_fif(fif_fname)
+    events = mne.find_events(raw)
+    event_id = {
+        'Auditory/Left': 1,
+        'Auditory/Right': 2,
+        'Visual/Left': 3,
+        'Visual/Right': 4,
+        'Visual/Smiley': 32,
+        'Motor/Button': 5
+    }
+    event_map = {v: k for k, v in event_id.items()}
+    annot = Annotations(onset=raw.times[events[:, 0] - raw.first_samp],
+                        duration=np.zeros(len(events)),
+                        description=[event_map[vv] for vv in events[:, 2]],
+                        orig_time=None)
+    raw.set_annotations(annot)
+
+    events2, event_id2 = \
+        events_from_annotations(raw, event_id=event_id, regexp=None)
+    assert_array_equal(events, events2)
+    assert_equal(event_id, event_id2)
+
+    events3, event_id3 = \
+        events_from_annotations(raw, event_id=None, regexp=None)
+
+    assert_array_equal(events[:, 0], events3[:, 0])
+    assert list(event_id.keys()).sort() == list(event_id3.keys()).sort()
```

Comment:
why do you need to sort? it should not be necessary

Reply:
Otherwise python 2 breaks.

Reply:
Or you need to use an ordered dict. And since the code does not need it, just sorting on testing avoids the overhead

Reply:
it's simpler to use `set(event_id.keys())` or `sorted(event_id.keys())` than `list(event_id.keys()).sort()`